llama-index-integrations/agent/llama-index-agent-openai/llama_index/agent/openai/step.py

```diff
@@ -195,6 +199,7 @@ def from_tools(
         callback_manager: Optional[CallbackManager] = None,
         system_prompt: Optional[str] = None,
         prefix_messages: Optional[List[ChatMessage]] = None,
+        get_tool_by: Callable[[List[BaseTool], str], BaseTool] = get_function_by_name,
```

Comment:
can we add a similar hook to the react agent? And maybe add a notebook example of how this should be used?

Reply:
LOL that's fair I guess. I could take a stab.

I would add to an existing example maybe... just adding a section showing this specific customization

Reply:
Could look into it if you want, it just looks a bit more complex